dtfs-central-api/src/scheduler/jobs/create-utilisation-reports/index.test.ts

```diff
@@ -0,0 +1,148 @@
+      jest.mocked(getAllBanks).mockResolvedValue([]);
+      const saveUtilisationReportDetailsSpy = jest.spyOn(utilisationReportsRepo, 'saveUtilisationReportDetails');
+
+      // Act
+      await job.task(new Date());
+
+      // Assert
+      expect(saveUtilisationReportDetailsSpy).not.toHaveBeenCalled();
+    });
+
+    it('does not try to create any utilisation reports when reports for all banks in the current period already exist', async () => {
+      // Arrange
+      const job = getJob();
+      const bank = { id: '123', name: 'Test bank' } as Bank;
+      jest.mocked(getAllBanks).mockResolvedValue([bank]);
+      const getCurrentUtilisationReportByBankIdAndReportPeriodSpy = jest
+        .mocked(utilisationReportsRepo.getCurrentUtilisationReportByBankIdAndReportPeriod)
+        .mockResolvedValue({} as UtilisationReport);
+      const saveNotReceivedUtilisationReportSpy = jest.spyOn(utilisationReportsRepo, 'saveNotReceivedUtilisationReport');
+
+      // Act
+      await job.task(new Date());
+
+      // Assert
+      expect(getCurrentUtilisationReportByBankIdAndReportPeriodSpy).toHaveBeenCalledWith(bank.id, mockReportPeriod);
+      expect(saveNotReceivedUtilisationReportSpy).not.toHaveBeenCalled();
+    });
+
+    const banks = [
+      { id: '1', name: 'Bank 1' },
+      { id: '2', name: 'Bank 2' },
+      { id: '3', name: 'Bank 3' },
+    ] as Bank[];
+
+    it('tries to create utilisation reports for all banks when reports for all banks in the current period do not exist', async () => {
+      // Arrange
+      const job = getJob();
+      jest.mocked(getAllBanks).mockResolvedValue(banks);
+      const getCurrentUtilisationReportByBankIdAndReportPeriodSpy = jest
+        .mocked(utilisationReportsRepo.getCurrentUtilisationReportByBankIdAndReportPeriod)
+        .mockResolvedValue(null);
+      const saveNotReceivedUtilisationReportSpy = jest.spyOn(utilisationReportsRepo, 'saveNotReceivedUtilisationReport');
+
+      // Act
+      await job.task(new Date());
+
+      // Assert
+      expect(getCurrentUtilisationReportByBankIdAndReportPeriodSpy).toHaveBeenCalledTimes(banks.length);
+      banks.forEach(({ id }) => expect(getCurrentUtilisationReportByBankIdAndReportPeriodSpy).toHaveBeenCalledWith(id, mockReportPeriod));
+      expect(saveNotReceivedUtilisationReportSpy).toHaveBeenCalledTimes(banks.length);
+      banks.forEach((bank) => expect(saveNotReceivedUtilisationReportSpy).toHaveBeenCalledWith(mockReportPeriod, bank));
+    });
+
+    it('only tries to create reports for banks which do not have a report for the current report period', async () => {
+      // Arrange
+      const job = getJob();
+      jest.mocked(getAllBanks).mockResolvedValue(banks);
+      const getCurrentUtilisationReportByBankIdAndReportPeriodSpy = jest
+        .mocked(utilisationReportsRepo.getCurrentUtilisationReportByBankIdAndReportPeriod)
+        .mockImplementation((bankId: string) => {
+          switch (bankId) {
+            case '1':
```

Comment:
Instead of hard-coding this here I would add the following above:
```
const bankWithoutReport = banks[0]
```
then you can do
```
case bankWithoutReport.id:
```
here, and
```
expect(saveNotReceivedUtilisationReportSpy).toHaveBeenCalledWith(mockReportPeriod, bankWithoutReport);
```
in the assertion below, then it's all self-explanatory.